x-pack/plugins/cloud_integrations/cloud_experiments/common/types.ts

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { FEATURE_FLAG_NAMES, METRIC_NAMES } from './constants';
+
+/**
+ * The contract of the setup lifecycle method.
+ */
+export interface CloudExperimentsPluginSetup {
+  /**
+   * Identifies the user in the A/B testing service.
+   * For now, we only rely on the user ID. In the future, we may request further details for more targeted experiments.
+   * @param userId The unique identifier of the user in the experiment.
+   * @param userMetadata Additional attributes to the user.
+   */
+  identifyUser: (
+    userId: string,
+    userMetadata?: Record<string, string | boolean | number | Array<string | boolean | number>>
+  ) => void;
+
+  /**
+   * Fetch the configuration assigned to variation `configKey`. If nothing is found, fallback to `defaultValue`.
+   * @param featureFlagName The name of the key to find the config variation. {@link CloudExperimentsFeatureFlagNames}.
+   * @param defaultValue The fallback value in case no variation is found.
+   */
+  getVariation: <Data>(
```

Comment:
Do we need to expose `getVariation` on `setup` at this time? If there isn't a need for it yet, my vote would be to leave it out of the `setup` lifecycle until we know it is required, and just start by exposing it on `start`

Reply:
Removed from the `setup` lifecycle for now.